src/pynxtools_raman/rod/rod_reader.py

```diff
@@ -0,0 +1,161 @@
+from typing import List, Dict, Any, Union
+import gemmi  # for cif file handling
+
+from pathlib import Path
+
+
+class RodParser:
+    """
+    This class provides the ultilieies to read in a .rod file with "get_cif_file_content".
+    Then extract all data via "extract_keys_and_values_from_cif" into a dictionary.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.cif_doc = None
+        self.cif_block = None
+        self.lines = []
+        self.n_headerlines = 12
+
+    def _read_lines(self, file: Union[str, Path]):
+        """
+        Read all lines from the input files.
+        """
+        with open(file, encoding="utf-8") as utf8_file:
+            lines = utf8_file.readlines()
+
+        return lines
+
+    def get_cif_file_content(self, file_path):
+        doc = gemmi.cif.read_file(file_path)
+        block = doc.sole_block()  # extract main block of cif file
+        self.cif_doc = doc
+        self.cif_block = block
+        self.lines = self._read_lines(file_path)
+
+    def get_string_position(self, string_element: str, check_only_pos_zero=False):
+        line_positions_of_str_element = []
+
+        rod_lines = self.lines
+        if check_only_pos_zero:
+            for line_number, lines in enumerate(rod_lines):
+                if string_element in lines[0]:
+                    line_positions_of_str_element.append(line_number)
+        else:
+            if rod_lines is None:
+                print("rod_lines_is_none")
```

Comment:
Maybe use a logger.warning? For this, you need to instantiate a logger in this file (like in the reader.py).

Reply:
yea, thanks. Did not take a look into this for some time.